tests/cupyx_tests/scipy_tests/stats_tests/test_stats.py

```diff
@@ -60,3 +64,173 @@ def test_propotion_too_big(self):
             a = xp.array([4, 8, 2, 0, 9, 5, 10, 1, 7, 3, 6])
             with pytest.raises(ValueError):
                 scp.stats.trim_mean(a, 0.6)
+
+
+@testing.with_requires('scipy>=1.7')
+class TestZmap:
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_zmap_1dim(self, xp, scp, dtype):
+        x = testing.shaped_random((10,), xp, dtype=dtype)
+        y = testing.shaped_random((8,), xp, dtype=dtype)
+        return scp.stats.zmap(x, y)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_zmap_2dim(self, xp, scp, dtype):
+        x = testing.shaped_random((2, 6), xp, dtype=dtype)
+        y = testing.shaped_random((2, 1), xp, dtype=dtype)
+        return scp.stats.zmap(x, y)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=atol, rtol=rtol)
+    def test_zmap_multi_dim(self, xp, scp, dtype):
+        x = testing.shaped_random((3, 4, 5, 7), xp, dtype=dtype)
+        y = testing.shaped_random((3, 4, 1, 1), xp, dtype=dtype)
+        return scp.stats.zmap(x, y)
+
+    @testing.for_all_dtypes(no_float16=True, no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=atol, rtol=rtol)
+    def test_zmap_multi_dim_2(self, xp, scp, dtype):
+        x = testing.shaped_random((4, 4, 5, 6, 2), xp, dtype=dtype)
+        y = testing.shaped_random((4, 4, 5, 6, 2), xp, dtype=dtype)
+        return scp.stats.zmap(x, y)
+
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-3, rtol=1e-3)
+    def test_zmap_multi_dim_2_float16(self, xp, scp):
+        x = testing.shaped_random((4, 4, 5, 6, 2), xp, dtype=xp.float16)
+        y = testing.shaped_random((4, 4, 5, 6, 2), xp, dtype=xp.float16)
+        return scp.stats.zmap(x, y)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=atol, rtol=rtol)
+    def test_zmap_with_axis(self, xp, scp, dtype):
+        x = testing.shaped_random((2, 3), xp, dtype=dtype)
+        y = testing.shaped_random((1, 3), xp, dtype=dtype)
+        return scp.stats.zmap(x, y, axis=1)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=atol, rtol=rtol)
+    def test_zmap_with_axis_ddof(self, xp, scp, dtype):
+        x = testing.shaped_random((4, 5), xp, dtype=dtype)
+        y = testing.shaped_random((1, 5), xp, dtype=dtype)
+        return scp.stats.zmap(x, y, axis=1, ddof=2)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_zmap_empty(self, xp, scp, dtype):
+        x = xp.array([], dtype=dtype)
+        y = xp.array([1, 3, 5], dtype=dtype)
+        return scp.stats.zmap(x, y)
+
+    @testing.for_dtypes('fdFD')
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=atol, rtol=rtol)
+    def test_zmap_nan_policy_propagate(self, xp, scp, dtype):
```

Comment:
We want to skip only tests that use `nan_policy`
```suggestion
    @testing.with_requires('scipy>=1.7')
    def test_zmap_nan_policy_propagate(self, xp, scp, dtype):
```

Reply:
Got that!